addons/mrp/models/mrp_production.py

```diff
@@ -867,15 +867,27 @@ def write(self, vals):
                 if production.state == 'draft' and picking_type != production.picking_type_id:
                     production.name = picking_type.sequence_id.next_by_id()
 
+        date_start_map = dict()
+        if 'date_start' in vals:
+            date_start_map = {
+                prod: vals['date_start'] - datetime.timedelta(days=prod.bom_id.produce_delay)
+                if prod.bom_id else vals['date_start']
+                for prod in self
+            }
+            del vals['date_start']
+
         res = super(MrpProduction, self).write(vals)
```

Comment:
I would rather write everything in batches rather than writing every vals without date_start and then start date in batches. Something like this perhaps:
```suggestion
if date_start_map:
    res = True
    for production, date_start in date_start_map.items():
        res &= super(MrpProduction, production).write(vals | {'date_start': date_start})
else:
     res = super(MrpProduction, self).write(vals)
```